src/handlers.ts

```diff
@@ -0,0 +1,29 @@
+import { ConstConstError } from "./errors";
+
+export const mutationHandler: ProxyHandler<any> = {
+    set(_, prop, val) {
+        throw ConstConstError.newSetError(prop, val);
+    },
+    deleteProperty(_, prop) {
+        throw ConstConstError.newDeleteError(prop);
+    }
+}
+
+export const mapMutationHandler: ProxyHandler<any> = {
+    get (tgt, prop, rcvr) {
+        if (prop === "size") {
+            return tgt.size;
+        }
+        let val = Reflect.get(tgt, prop, rcvr);
+        if (prop === "set" && typeof val === "function") {
+            throw ConstConstError.newMapSetError();
+        } else if (prop === "clear" && typeof val === "function") {
+            throw ConstConstError.newMapClearError();
+        } else if (prop === "delete" && typeof val === "function") {
```

Comment:
typeof checks seems excessive here. if prop is going to one of set, clear, delete and given that this handler is only applied to maps should not cause any issue even if we don't check for functions.